File: aliyun-java-best-practise/src/main/java/com/aliyun/bestpractise/DescribeInstancesDemo.java
package com.aliyun.bestpractise;

import com.aliyuncs.DefaultAcsClient;
import com.aliyuncs.IAcsClient;
import com.aliyuncs.ecs.model.v20140526.DescribeInstancesRequest;
import com.aliyuncs.ecs.model.v20140526.DescribeInstancesResponse;
import com.aliyuncs.exceptions.ClientException;
import com.aliyuncs.profile.DefaultProfile;

public class DescribeInstancesDemo {
    public static void main(String[] args) {
        DefaultProfile profile = DefaultProfile.getProfile(
                "RegionId",
                "accessKeyId",
                "accessKeySecret");
        IAcsClient client = new DefaultAcsClient(profile);
        DescribeInstancesRequest request = new DescribeInstancesRequest();
        DescribeInstancesResponse response;
        try {
            response = client.getAcsResponse(request);
            System.out.println(response);
        } catch (ClientException e) {
            e.printStackTrace();
        }
    }
}

File: aliyun-python-best-practise/aliyunbestpractise/describeInstances_demo.py
from aliyunsdkcore.client import AcsClient
from aliyunsdkecs.request.v20140526.DescribeInstancesRequest import DescribeInstancesRequest
client = AcsClient('access_key_id', 'access_key_secret', 'region_id')
request = DescribeInstancesRequest()
response = client.do_action_with_exception(request)
print(response)